src/scmrepo/git/backend/pygit2.py

```diff
@@ -421,7 +472,58 @@ def fetch_refspecs(
         progress: Callable[["GitProgressEvent"], None] = None,
         **kwargs,
     ) -> Mapping[str, SyncStatus]:
-        raise NotImplementedError
+        from pygit2 import GitError
+
+        if isinstance(refspecs, str):
+            refspecs = [refspecs]
+
+        with self.get_remote(url) as remote:
+            if os.name == "nt" and remote.url.startswith("ssh://"):
+                raise NotImplementedError
+
+            if os.name == "nt" and remote.url.startswith("file://"):
+                url = remote.url[len("file://") :]
+                self.repo.remotes.set_url(remote.name, url)
+                remote = self.repo.remotes[remote.name]
```

Comment:
Is this still needed even after the `remotes.create()` change?

Reply:
Looks like it still fails.

Reply:
You can make this change in the PR then, but we probably need to open another issue for it in scmrepo (this likely means that a regular git remote with `file://` URL in it is also broken for us on windows). This may still be related to us not setting the path separators or formatting the URL in a way that libgit2 expects and probably needs some more investigation.